frontend/src/pages/Main/index.tsx

```diff
@@ -1,11 +1,99 @@
-import { Component } from "solid-js";
-import Hello from "../../components/Hello";
-import { Typography } from "@suid/material";
-
-const Main: Component = () => {
-  return <>
-    <Hello />
-  </>
+import { Container, AppBar, Toolbar, Box, TextField, useTheme, Grid, CardMedia } from "@suid/material"
+import { Component, For, createEffect, createSignal, onCleanup } from "solid-js"
+import GoodCard from "../../components/GoodCard"
+import { createAsync } from "@solidjs/router"
+import { getGood, getGoods } from "../../lib/store"
+import { transform } from "@suid/vite-plugin"
+
+interface GridGoodCardProps {
+  id: number
+}
+
+interface CarouselProps {
+  ids: number[]
+}
+
+// CSS 动画样式
+const boxSx = {
+};
+
+const Carousel: Component<CarouselProps> = ({ ids }) => {
+  const [currentIndex, setCurrentIndex] = createSignal(0);
+  const [isLoaded, setIsLoaded] = createSignal(false);
+
+  const nextImage = () => {
+    setIsLoaded(false);
+    setCurrentIndex((prevIndex) => (prevIndex + 1) % ids.length);
+  };
+
+  const prevImage = () => {
+    setIsLoaded(false);
+    setCurrentIndex((prevIndex) => (prevIndex - 1 + ids.length) % ids.length);
+  };
+
+  const good = createAsync(() => getGood(ids[currentIndex()]));
+
+  // 自动翻页
+  createEffect(() => {
+    const intervalId = setInterval(nextImage, 3000); // 每3秒翻页一次
+    onCleanup(() => clearInterval(intervalId)); // 组件卸载时清除定时器
+  });
```

Comment:
maybe should use [onMount](https://docs.solidjs.com/concepts/effects#onmount)?

Reply:
Let me have a try (